src/borg/repository.py

```diff
@@ -1504,8 +1511,9 @@ def iter_objects(self, segment, offset=0, include_data=False, read_data=True):
             # we are touching this segment for the first time, check the MAGIC.
             # Repository.scan() calls us with segment > 0 when it continues an ongoing iteration
             # from a marker position - but then we have checked the magic before already.
-            if fd.read(MAGIC_LEN) != MAGIC:
-                raise IntegrityError(f'Invalid segment magic [segment {segment}, offset {0}]')
+            mr = fd.read(MAGIC_LEN)
+            if mr != MAGIC:
+                raise IntegrityError(f'Invalid segment magic [segment {segment}, offset {0}, magic {mr!r}]')
```

Comment:
one test fails here with `mr == b''`.

Reply:
does anybody see why?

Reply:
https://github.com/borgbackup/borg/runs/6403364221?check_suite_focus=true#step:9:3258

Reply:
guess i found it, see commits.